### PR TITLE
feat(share): add the share operator

### LINKED ISSUE #439: Implement `share` operator
This is an alias for `publish().refCount()`, but is honestly preferable for simplicity's sake. It's easier to teach people `share()` than `publish().refCount()`, because to a newbie, the latter won't make much sense.

Ideally, this will be implemented by pulling in the `publish` method via import, rather than depending on it being on the Observable. Since this is a modular library, there is no guarantee publish has been added.

### PR DESCRIPTION
**Pull request by blesh**
closes #439

### CODE REVIEW

**Comment by staltz** [comment]
I would merge :+1:

File: spec/operators/share-spec.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    var source = obs.share();
+    
+    expect(subscriptionCount).toBe(0);
+    
+    source.subscribe();
+    source.subscribe();
+    
+    expect(subscriptionCount).toBe(1);
+  });
+
```
Comment (staltz) [inline spec/operators/share-spec.js]: Should we also have a marble test where `e1` is a cold Observable to make sense applying `share` on it? Also the case of multiple subscribers to a was-cold shared Observable makes a lot of sense to have a subscribe-and-unsubscribe marble diagram test that we mentioned in #428.

File: perf/micro/immediate-scheduler/operators/share.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        .add('old share with immediate scheduler', function () {
+            oldShareWithImmediateScheduler.subscribe(_next, _error, _complete);
+        })
+        .add('new share with immediate scheduler', function () {
+            newShareWithImmediateScheduler.subscribe(_next, _error, _complete);
+        });
+
+    function add(acc, x) {
+        return x + x
+    }
```
Comment (staltz) [inline perf/micro/immediate-scheduler/operators/share.js]: `add` is not used.

File: spec/operators/share-spec.js (hunk 1 above)
Comment (blesh) [inline spec/operators/share-spec.js]: Probably

File: perf/micro/immediate-scheduler/operators/share.js (hunk 2 above)
Comment (blesh) [inline perf/micro/immediate-scheduler/operators/share.js]: heh... cruft.

**Pull request merged**